packages/reflector/README.md

```diff
@@ -446,274 +477,398 @@ shapeReflector.getAccessor("color").addDecorator(accessors());
+
+**Example with `autoSync`:**
+
+```ts
+// Reflector with autoSync enabled
+const autoSyncReflector: Reflector = Reflector.from(SomeClass, true);
+```
+
+> **Note:** For more information on dynamic decoration, refer to the [Dynamic Decoration](#dynamic-decoration) section.
 
 ### Reflected Types
 
-There are four class member types and one parameter type, that Reflector can acquire:
+The Reflector API can access four types of class members and one parameter type:
 
 ```ts
-enum ClassMemberType {
-    CONSTRUCTOR = "CONSTRUCTOR",
-    PROPERTY = "PROPERTY",
-    ACCESSOR = "ACCESSOR",
-    METHOD = "METHOD",
+//---PRIMITIVE DECORATED ELEMENTS---------------------------------------------------------------------------
+interface PrimitiveDecoratedElementType {
+    CONSTRUCTOR: number
+    PROPERTY: number;
+    ACCESSOR: number;
+    METHOD: number;
+    CONSTRUCTOR_PARAMETER: number;
+    METHODS_PARAMETER: number;
 }
 
-enum ParameterType {
-    PARAMETER = "PARAMETER",
+const PrimitiveDecoratedElementEnum: Readonly<PrimitiveDecoratedElementType> = Object.freeze({
+    CONSTRUCTOR: 1,
+    PROPERTY: 2,
+    ACCESSOR: 4,
+    METHOD: 8,
+    CONSTRUCTOR_PARAMETER: 16,
+    METHODS_PARAMETER: 32,
+});
+
+//---ADVANCED DECORATED ELEMENTS---------------------------------------------------------------------------
+export interface DecoratedElementType extends PrimitiveDecoratedElementType {
+    CLASS_MEMBER: number;
+    DECORATED_ELEMENT: number;
+    EXECUTABLE_ELEMENT: number;
+    FIELD: number;
+    PARAMETER: number;
 }
+
+export const DecoratedElementEnum: Readonly<DecoratedElementType> = Object.freeze({
+    CONSTRUCTOR: PrimitiveDecoratedElementEnum.CONSTRUCTOR,
+    
+  	METHOD: PrimitiveDecoratedElementEnum.METHOD,
+    
+  	EXECUTABLE_ELEMENT: PrimitiveDecoratedElementEnum.METHOD | 
+        PrimitiveDecoratedElementEnum.CONSTRUCTOR,
+
+    PROPERTY: PrimitiveDecoratedElementEnum.PROPERTY,
+    
+  	ACCESSOR: PrimitiveDecoratedElementEnum.ACCESSOR,
+   
+  	FIELD: PrimitiveDecoratedElementEnum.PROPERTY | 
+        PrimitiveDecoratedElementEnum.ACCESSOR,
+
+    CLASS_MEMBER: PrimitiveDecoratedElementEnum.PROPERTY | 
+        PrimitiveDecoratedElementEnum.ACCESSOR | 
+        PrimitiveDecoratedElementEnum.METHOD | 
+        PrimitiveDecoratedElementEnum.CONSTRUCTOR,
+
+    CONSTRUCTOR_PARAMETER: PrimitiveDecoratedElementEnum.CONSTRUCTOR_PARAMETER,
+    
+  	METHODS_PARAMETER: PrimitiveDecoratedElementEnum.METHODS_PARAMETER,
+    
+  	PARAMETER: PrimitiveDecoratedElementEnum.CONSTRUCTOR_PARAMETER | 
+        PrimitiveDecoratedElementEnum.METHODS_PARAMETER,
+
+    DECORATED_ELEMENT: PrimitiveDecoratedElementEnum.PROPERTY | 
+        PrimitiveDecoratedElementEnum.ACCESSOR | 
+        PrimitiveDecoratedElementEnum.METHOD | 
+        PrimitiveDecoratedElementEnum.CONSTRUCTOR | 
+        PrimitiveDecoratedElementEnum.CONSTRUCTOR_PARAMETER | 
+        PrimitiveDecoratedElementEnum.METHODS_PARAMETER,
+});
```

Comment:
I think there is an indentation problem. Can we make those all on the same level vertically?